canary/webrtc-c/jobs/runner.groovy

```diff
@@ -272,8 +262,28 @@ pipeline {
         string(name: 'GIT_HASH')
         booleanParam(name: 'FIRST_ITERATION', defaultValue: true)
     }
+    
+    // Set the role ARN to environment to avoid string interpolation to follow Jenkins security guidelines.
+    environment {
+        AWS_KVS_STS_ROLE_ARN = credentials('CANARY_STS_ROLE_ARN')
+    }
 
     stages {
+        stage('Fetch STS credentials and export to env vars') {
+            steps {
+                script {
+                    def assumeRoleOutput = sh(script: 'aws sts assume-role --role-arn $AWS_KVS_STS_ROLE_ARN --role-session-name roleSessionName --duration-seconds 43200 --output json',
+                                                returnStdout: true
+                                                ).trim()
+                    def assumeRoleJson = readJSON text: assumeRoleOutput
+
```

Comment:
check the response contains AK  (valid)?

Reply:
The Canary will fail with the following log if the AKs are not set:

`
2025-02-19 22:38:23.986 ERROR   mustenv(): AWS_ACCESS_KEY_ID must be set
`